src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -939,3 +957,222 @@ void AppHost::_SummonWindowRequested(const winrt::Windows::Foundation::IInspecta
     summonArgs.ToggleVisibility(false); // Do not toggle, just make visible.
     _HandleSummon(sender, summonArgs);
 }
+
+void AppHost::_MinimizeToTrayRequested(const winrt::Windows::Foundation::IInspectable&,
+                                       const winrt::Windows::Foundation::IInspectable&)
+{
+    _window->HideWindow();
+}
+
+void AppHost::_HandleTrayIconPressed()
+{
+    // No name in the args means summon the mru window.
+    Remoting::SummonWindowSelectionArgs args{};
+    args.SummonBehavior().ToggleVisibility(false);
+    _windowManager.SummonWindow(args);
+}
+
+// Method Description:
+// - Creates and adds an icon to the notification tray.
+// If an icon already exists, update the HWND associated
+// to the icon with this window's HWND.
+// Arguments:
+// - <unused>
+// Return Value:
+// - <none>
+void AppHost::_UpdateTrayIcon()
+{
+    if (_logic.GetAlwaysShowTrayIcon() || _logic.GetMinimizeToTray())
+    {
+        if (!_trayIconData)
+        {
+            NOTIFYICONDATA nid{};
+            nid.cbSize = sizeof(NOTIFYICONDATA);
+
+            // This HWND will receive the callbacks sent by the tray icon.
+            nid.hWnd = _window->GetHandle();
+
+            // App-defined identifier of the icon. The HWND and ID are used
+            // to identify which icon to operate on when calling Shell_NotifyIcon.
+            // Multiple icons can be associated with one HWND, but here we're only
+            // going to be showing one so the ID doesn't really matter.
+            nid.uID = 1;
+
+            nid.uCallbackMessage = CM_NOTIFY_FROM_TRAY;
+
+            nid.hIcon = static_cast<HICON>(GetActiveAppIconHandle(true));
+            StringCchCopy(nid.szTip, ARRAYSIZE(nid.szTip), RS_(L"AppName").c_str());
+            nid.uFlags = NIF_MESSAGE | NIF_SHOWTIP | NIF_TIP | NIF_ICON;
+            Shell_NotifyIcon(NIM_ADD, &nid);
+
+            // For whatever reason, the NIM_ADD call doesn't seem to set the version
+            // properly, resulting in us being unable to receive the expected notification
+            // events. We actually have to make a separate NIM_SETVERSION call for it to
+            // work properly.
+            nid.uVersion = NOTIFYICON_VERSION_4;
+            Shell_NotifyIcon(NIM_SETVERSION, &nid);
+
+            _trayIconData = nid;
+        }
+    }
+    else if (_trayIconData)
+    {
+        // We have a tray icon existing, but the now the new settings
+        // are telling us we don't want you so poof.

```

Comment:
This sentence doesn't read right

Reply:
I tried too hard to meme here, should've went back and revised my comments 😆